File: src/main/java/com/mohe/spring/repository/UserPreferenceVectorRepository.java
package com.mohe.spring.repository;

import com.mohe.spring.entity.UserPreferenceVector;
import org.springframework.data.jpa.repository.JpaRepository;
import org.springframework.stereotype.Repository;

import java.util.List;
import java.util.Optional;

@Repository
public interface UserPreferenceVectorRepository extends JpaRepository<UserPreferenceVector, Long> {
    
    Optional<UserPreferenceVector> findByUserId(Long userId);
    
    void deleteByUserId(Long userId);
    
    List<UserPreferenceVector> findByUserIdIn(List<Long> userIds);
}

File: src/main/kotlin/com/mohe/spring/repository/UserPreferenceVectorRepository.kt
package com.mohe.spring.repository

import com.mohe.spring.entity.UserPreferenceVector
import org.springframework.data.jpa.repository.JpaRepository
import org.springframework.stereotype.Repository

@Repository
interface UserPreferenceVectorRepository : JpaRepository<UserPreferenceVector, Long> {
    fun findByUserId(userId: Long): UserPreferenceVector?
    fun deleteByUserId(userId: Long)
    fun findByUserIdIn(userIds: List<Long>): List<UserPreferenceVector>
}